### PR TITLE
Scons: provide the number of logical processors as the number of maximum concurrent jobs to speed up builds

### PR DESCRIPTION
**Pull request by LeonarddeR**
### Link to issue number:
None

### Summary of the issue:
Building NVDA can take some time. SCons has the ability to provide the number of concurrent jobs during build.

### Description of how this pull request fixes the issue:
Specify the maximum number of concurrent jobs when building NVDA.

### Testing strategy:
Tested locally on an Intel Core i7-9850H using PowerShell's `measure-command { .\scons.bat source }`, ensuring the venv was created prerun.

* Without the change, 170 seconds
* With the change, 59 seconds

### Known issues with pull request:
Output sometimes get a bit scrambled, i.e. the output of the cl command comes before SCons output the name of the file being build.

### Change log entries:
For Developers
- Set up SCons to build with multiple concurrent jobs, equal to the number of logical processors in the system (i.e. 8 on a quat core Intel CPU with hyper threading). This can dramatically decrease build times on multi core systems.

### Code Review Checklist:

- [x] Pull Request description:
  - description is up to date
  - change log entries
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] API is compatible with existing add-ons.
- [x] Documentation:
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English

### CODE REVIEW

**Comment by leonardder** [comment]
Note, this doesn't seem to have any noticeable benefit on appveyor.

**Comment by josephsl** [comment]
Hi, probably because it uses virtual machine instances, which can impact processor performance. Thanks.

**Comment by Brian1Gaff** [comment]
So if that failed would the correct errors still occur, Not thatI would be 
doing this, but taking up the whole of the processor in this way is bound to 
slug bits of it?
 Brian

***@***.***
Sent via blueyonder.
Please address personal E-mail to:-
***@***.***, putting 'Brian Gaff'
in the display name field.
Newsgroup monitored: alt.comp.blind-users
----- Original Message ----- 
From: "Leonard de Ruijter" ***@***.***>
To: "nvaccess/nvda" ***@***.***>
Cc: "Subscribed" ***@***.***>
Sent: Monday, January 10, 2022 6:50 PM
Subject: [nvaccess/nvda] Scons: provide the number of logical processors as 
the number of maximum concurrent jobs to speed up builds (PR #13226)

**Comment by seanbudd** [comment]
Is there a way to disable this for appveyor? Scrambling the output of try/PR builds and official builds without a time improvement seems like a loss.

**Review by seanbudd — APPROVED** [review]
Thanks @leonardder

**Comment by michaelDCurran** [comment]
I'm really struggling with the merging of this pr. It is very hard to follow build output now, especially when trying to see errors from cl. This gets worse with the introduction of calling msbuild for the remote ops library.
Note that I have 12 logica processors, so there is a great deal going on at the same time.
I think parallel builds are a great advantage, but I do feel they should be optional. I.e. by specifying -j 12 (or how ever many you want) when running scons.
Pr #13367 has become necessary since this merge. And although we should probably take that one as parallel builds should work if people want them, I don't believe enforcing them by default, especially within a developer context is the right thing here. 

I feel that this pr should be reverted.

**Comment by seanbudd** [comment]
@leonardder @michaelDCurran  can this be overridden with `-j1` eg `scons source -j1`? This could either be a workaround, at least until the PR is reverted. 

Would you like me to revert this? GitHub won't open a revert PR automatically

**Comment by lukaszgo1** [comment]
Have you reported the scrambled output thing to SCons?? While this would probably be pretty difficult to fix not reporting the problem makes it guaranteed not to be fixed.
